src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -397,6 +397,9 @@ void ImageReplayer<I>::handle_bootstrap(int r) {
     m_delete_requested = true;
     on_start_fail(0, "remote image no longer exists");
     return;
+  } else if (r == -ERESTART) {
+    restart(nullptr);
```

Comment:
Do we need it? When in my suggestion I was talking about restart, I meant that we would make sure that when we got `-ERESTART` here, we didn't set `m_finished` flag, so the instance replayer would try to start it again later. I.e. the same way as we do for other not "fatal" errors. 

I think w just need  `on_start_fail` with some useful error description like "image in transient state, try again".

Reply:
I did this because I thought that this issue could appear when the ImageReplayer is restarted by the InstanceReplayer to be removed so it would be his last chance to be restarted. But I think this case might never happen anyway.

Let's try you way first and try to see if it's ok.